pydantic_ai_slim/pydantic_ai/tools.py

```diff
@@ -9,16 +10,34 @@
 from opentelemetry.trace import Tracer
 from pydantic import ValidationError
 from pydantic.json_schema import GenerateJsonSchema, JsonSchemaValue
-from pydantic_core import core_schema
-from typing_extensions import Concatenate, ParamSpec, TypeAlias, TypeVar
+from pydantic_core import SchemaValidator, core_schema
+from typing_extensions import Concatenate, ParamSpec, Self, TypeAlias, TypeVar
 
 from . import _function_schema, _utils, messages as _messages
 from .exceptions import ModelRetry, UnexpectedModelBehavior
 
 if TYPE_CHECKING:
+    from typing import Protocol
+
     from .models import Model
     from .result import Usage
 
+    class LangChainTool(Protocol):
+        # args are like
+        # {'dir_path': {'default': '.', 'description': 'Subdirectory to search in.', 'title': 'Dir Path', 'type': 'string'},
+        #  'pattern': {'description': 'Unix shell regex, where * matches everything.', 'title': 'Pattern', 'type': 'string'}}
+        @property
+        def args(self) -> dict[str, dict[str, str]]: ...
```

Comment:
This is an example with 2 string arguments, but if there are dict or list arguments, I imagine this could be more deeply nested and include value other than strings. Could we just make this `dict[str, Any]` to support any JSON schema?